plugin/src/main/java/org/wildfly/swarm/plugin/gradle/GradleArtifactResolvingHelper.java

```diff
@@ -122,22 +110,19 @@ public ArtifactSpec resolve(final ArtifactSpec spec) {
                     }
                 });
 
-        if ( transitive ) {
-            return config.getResolvedConfiguration().getFirstLevelModuleDependencies()
-                    .stream()
-                    .flatMap(this::fullTree)
-                    .collect(Collectors.toSet());
+        if (transitive) {
+            return config
+                .getResolvedConfiguration()
+                .getResolvedArtifacts();
         }
-        return config.getResolvedConfiguration().getFirstLevelModuleDependencies();
-    }
-
-    private Stream<ResolvedDependency> fullTree(ResolvedDependency root) {
-        return Stream.concat( Stream.of( root ), subTree( root ) );
-    }
 
-    private Stream<ResolvedDependency> subTree(ResolvedDependency root) {
-        return root.getChildren().stream()
-                .flatMap( e-> fullTree( e ));
+        return config
+            .getResolvedConfiguration()
+            .getFirstLevelModuleDependencies()
+            .stream()
+            .map(dep -> dep.getModuleArtifacts())
+            .flatMap(artifacts -> artifacts.stream())
+            .collect(Collectors.toList());
```

Comment:
Gradle will not return duplicates then resolving all the artifacts for the resolved dependency in a previous step. this step from dependency to module-artifact is mostly a 1:1 conversion - because there aren't many modules with multiple artifacts.